### PR TITLE
Copy function

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

Small fun feature for copying a function.

### Test Plan

unit tests

### CODE REVIEW

File: assets/panel/debugger.properties [hunk 1]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
 copySourceUrl=Copy Source Url
 copySourceUrl.accesskey=u
 
+# LOCALIZATION NOTE (copyFunctionUrl): This is the text that appears in the
```
Comment (flodolo) [inline assets/panel/debugger.properties]: Wrong string reference and comment

File: assets/panel/debugger.properties [hunk 2]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
...
+# context menu to copy the source URL of file open.
+copyFunction=Copy Function
+copyFunction.accesskey=f
```
Comment (flodolo) [inline assets/panel/debugger.properties]: F

File: assets/panel/debugger.properties (hunk 1 above)
Comment (jasonLaster) [reply assets/panel/debugger.properties]: thanks.

**Comment by bomsy** [comment]
LGTM! not tested yet.

File: src/components/Editor/EditorMenu.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { showMenu } from "devtools-launchpad";
 import { isOriginalId } from "devtools-source-map";
 import { copyToTheClipboard } from "../../utils/clipboard";
+import { getClosestFunction } from "../../utils/breakpoint";
```
Comment (bomsy) [inline src/components/Editor/EditorMenu.js]: This does not seem to be used

File: src/utils/breakpoint/index.js [hunk 4]
```diff
@@ -3,7 +3,11 @@
 import { isEnabled } from "devtools-config";
 import { getBreakpoint } from "../../selectors";
 import assert from "../assert";
-export { getASTLocation, findScopeByName } from "./astBreakpointLocation";
+export {
+  getASTLocation,
+  findScopeByName,
+  getClosestFunction
```
Comment (bomsy) [inline src/utils/breakpoint/index.js]: this is not used any where

File: assets/panel/debugger.properties [hunk 5]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
 copySourceUrl=Copy Source Url
 copySourceUrl.accesskey=u
 
+# LOCALIZATION NOTE (copyFunction): This is the text that appears in the
+# context menu to copy the source URL of file open.
```
Comment (flodolo) [inline assets/panel/debugger.properties]: The comment is still wrong…

File: assets/panel/debugger.properties [hunk 6]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
...
+copyFunction=Copy Function
```
Comment (flodolo) [inline assets/panel/debugger.properties]: copyFunction -> copyFunction.label

**Comment by darkwing** [comment]
Small suggestion: "flash" the entire function as a way to signal to the user the function was copied?  Some sort of feedback would be awesome.

**Comment by jasonLaster** [comment]
@darkwing lets do that as a follow up. We have some great flash technology :P

File: assets/panel/debugger.properties [hunk 7]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
 copySourceUrl=Copy Source URL
 copySourceUrl.accesskey=u
 
+# LOCALIZATION NOTE (copyFunction): This is the text that appears in the
+# context menu to copy the function the user selected
+copyFunction.label=Copy Function
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: @flodolo sorry, that's embarrassing :/

**Comment by jasonLaster** [comment]
i guess this will need some more love...

**Comment by darkwing** [comment]
This is good to go IMO.  I agree the flash can come later.

File: assets/panel/debugger.properties (hunk 1 above)
Comment (juliandescottes) [reply assets/panel/debugger.properties]: Again this was not fixed before merging @jasonLaster

**Pull request merged**